src/main/kotlin/dev/arbjerg/ukulele/command/HelpCommand.kt

```diff
@@ -1,15 +1,35 @@
 package dev.arbjerg.ukulele.command
 
+import dev.arbjerg.ukulele.config.BotProps
 import dev.arbjerg.ukulele.features.HelpContext
 import dev.arbjerg.ukulele.jda.Command
 import dev.arbjerg.ukulele.jda.CommandContext
+import net.dv8tion.jda.api.MessageBuilder
 import org.springframework.stereotype.Component
 
 @Component
 class HelpCommand : Command("help") {
-    override suspend fun CommandContext.invoke() {
-        val command = beans.commandManager[argumentText.trim()] ?: return replyHelp()
-        replyHelp(command)
+    override suspend fun CommandContext.invoke() = when {
+        argumentText.isNotBlank() -> {
+            val command = beans.commandManager[argumentText.trim()]
+            if (command == null) {
+                replyHelp()
+            } else {
+                replyHelp(command)
+            }
+        }
+        else -> {
+            val commands = beans.commandManager.getCommands()
+            val msg = MessageBuilder()
+                .append("Available commands:")
+                .appendCodeBlock(buildString {
+                    commands.forEach {
```

Comment:
The commands value is unnecessary and you can just move it inline here
```suggestion
                    beans.commandManager.getCommands().forEach {
```

Reply:
Done